buildSrc/src/main/java/org/elasticsearch/gradle/precommit/DependencyLicensesTask.java

```
@@ -0,0 +1,334 @@
+        }
+
+        // TODO: shouldn't have to trim, sha files should not have trailing newline
+        byte[] fileBytes = Files.readAllBytes(shaFile.toPath());
+        String expectedSha = new String(fileBytes, StandardCharsets.UTF_8).trim();
+
+        String sha = getSha1(jar);
+
+        if (expectedSha.equals(sha) == false) {
+            throw new GradleException(
+                "SHA has changed! Expected " + expectedSha + " for " + jarName + " but got " + sha + ". " +
+                    "\nThis usually indicates a corrupt dependency cache or artifacts changed upstream." +
+                    "\nEither wipe your cache, fix the upstream artifact, or delete " + shaFile + " and run updateShas");
+        }
+        shaFiles.remove(shaFile);
+    }
+
+    private void checkFile(String name, String jarName, Map<String, Boolean> counters, String type) {
+        String fileName = getFileName(name, counters, type);
+
+        if (counters.containsKey(fileName) == false) {
+            throw new GradleException("Missing " + type + " for " + jarName + ", expected in " + fileName);
+        }
+
+        counters.put(fileName, true);
+    }
+
+    private String getFileName(String name, Map<String, ?> counters, String type) {
+        String fileName = name + "-" + type;
+
+        if (counters.containsKey(fileName) == false) {
+            // try the other suffix...TODO: get rid of this, just support ending in .txt
+            return fileName + ".txt";
+        }
+
+        return fileName;
+    }
+
+    @Input
+    public Map<String, String> getMappings() {
+        return new HashMap<>(mappings);
+    }
+
+    File getShaFile(String jarName) {
+        return new File(licensesDir, jarName + SHA_EXTENSION);
+    }
+
+    Set<File> getShaFiles() {
+        File[] array = licensesDir.listFiles();
+        if (array == null) {
+            return Collections.emptySet();
```

licensesDir should always be a directory, so we should throw an exception if we get a   null here .